react/features/connection-indicator/statsEmitter.js

```diff
@@ -0,0 +1,174 @@
+
+            if (framerateValue) {
+                this.remoteStatsUpdated(id, {
+                    framerate: framerateValue
+                });
+            }
+        });
+    },
+
+    /**
+     * Emit a stat update to all those listening for a specific user's
+     * connection stats.
+     *
+     * @param {string} id - The user id the stats are associated with.
+     * @param {Object} stats - New connection stats for the user.
+     * @returns {void}
+     */
+    remoteStatsUpdated(id, stats = {}) {
+        const callbacks = subscribers[id] || [];
+
+        callbacks.forEach(callback => {
+            callback(stats);
+        });
+    },
+
+    /**
+     * Have {@code statsEmitter} subscribe to stat updates from a given
+     * conference.
+     *
+     * @param {JitsiConference} conference - The conference for which
+     * {@code statsEmitter} should subscribe for stat updates.
+     * @returns {void}
+     */
+    startListeningForStats(conference) {
+        const { connectionQuality } = JitsiMeetJS.events;
+
+        conference.on(connectionQuality.LOCAL_STATS_UPDATED,
+            stats => this.localStatsUpdated(stats));
+
+        conference.on(connectionQuality.REMOTE_STATS_UPDATED,
+            (id, stats) => this.remoteStatsUpdated(id, stats));
+    },
+
+    /**
+     * Add a subscriber to be notified of stats updates for the local user.
+     *
+     * @param {Function} callback - The function to invoke when local stats have
+     * been updated.
+     * @returns {void}
+     */
+    subscribeToLocalStats(callback) {
+        this.subscribeToRemoteStats(LOCAL_USER_ID, callback);
+    },
+
+    /**
+     * Add a subscriber to be notified when stats are updated for a specified
+     * user id.
+     *
+     * @param {string} id - The user id whose stats updates are of interest.
+     * @param {Function} callback - The function to invoke when stats for the
+     * user have been updated.
+     * @returns {void}
+     */
+    subscribeToRemoteStats(id, callback) {
+        if (!subscribers[id]) {
+            subscribers[id] = [];
+        }
+
+        subscribers[id].push(callback);
+    },
+
+    /**
+     * Remove a subscriber that is listening for stats updates for the local
+     * user.
+     *
+     * @param {Function} callback - The function that is currently subscribed to
+     * stat updates for the local user.
+     * @returns {void}
+     */
+    unsubscribeFromLocalStats(callback) {
+        this.unsubscribeFromLocalStats(LOCAL_USER_ID, callback);
+    },
+
+    /**
+     * Remove a subscriber that is listening for stats updates for a specified
+     * user id.
+     *
+     * @param {string} id - The user id whose stats updates are no longer of
+     * interest.
+     * @param {Function} callback - The function that is currently subscribed to
+     * stat updates for the specified user id.
+     * @returns {void}
+     */
+    unsubscribeFromRemoteStats(id, callback) {
```

Comment:
Okay, what I did end up doing was having statsEmitter no-op if undefined is passed in. Then when the multi-user conference is joined, I have VideoLayout update the local video's connection indicator so that it can get the new userID. That way the local and remote methods in statsEmitter can be combined.